public/app/features/explore/utils/ResultProcessor.ts

```diff
@@ -16,23 +16,48 @@ import { getGraphSeriesModel } from 'app/plugins/panel/graph2/getGraphSeriesMode
 import { config } from 'app/core/config';
 
 export class ResultProcessor {
+  graphFrames: DataFrame[] = [];
+  tableFrames: DataFrame[] = [];
+  logsFrames: DataFrame[] = [];
+  traceFrames: DataFrame[] = [];
+
   constructor(
     private state: ExploreItemState,
     private dataFrames: DataFrame[],
     private intervalMs: number,
     private timeZone: TimeZone
-  ) {}
+  ) {
+    this.classifyFrames();
+  }
 
-  getGraphResult(): GraphSeriesXY[] | null {
-    const onlyTimeSeries = this.dataFrames.filter(frame => isTimeSeries(frame, this.state.datasourceInstance?.meta.id));
-    const timeSeriesToShowInGraph = onlyTimeSeries.filter(frame => shouldShowInVisualisationType(frame, 'graph'));
+  private classifyFrames() {
+    for (const frame of this.dataFrames) {
+      if (isTimeSeries(frame, this.state.datasourceInstance?.meta.id)) {
+        if (shouldShowInVisualisationType(frame, 'graph')) {
+          this.graphFrames.push(frame);
+        }
+        if (shouldShowInVisualisationType(frame, 'table')) {
+          this.tableFrames.push(frame);
+        }
+      } else {
+        if (shouldShowInVisualisationType(frame, 'logs')) {
```

Comment:
Do we need to nest these if/else-if statements inside the else statement starting at line 42 or can we just use all else-if statements instead?

Reply:
the thing is time series by default are shown in both graph and table. So they need to be separate, other types are sort of priority based in the sense that when we think something should be logs we won't try to match this to anything else.

Reply:
Also this assumes we will start using the meta attributes so may not work at the moment for all cases, but as I mentioned if we can do some sensible default that should be fine.